.circleci/config.yml

```diff
@@ -60,6 +60,29 @@ jobs:
     environment:
       <<: *ENVIRONMENT
 
+  # checks vendor directory is correct
+  check-vendor:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      <<: *ENVIRONMENT
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+          - consul-modcache-v1-{{ checksum "go.mod" }}
```

Comment:
Should this be `checksum "go.sum"` rather than `go.mod` keying the cache contents against the resolved hashes rather than desired versions? Less familiar with Go but for example in Node.js I'd typically key the CirlceCI cache against something like `package-lock.json` instead of `package.json`.

Reply:
This is one that I had a hard time wrapping my head around when modules first came out too. According to https://github.com/golang/go/wiki/Modules#is-gosum-a-lock-file-why-does-gosum-include-information-for-module-versions-i-am-no-longer-using `go.mod` is all you need for reproducible builds. The `go.sum` file is there for checksum validation. My understanding is that because it contains checksums for whatever versions you have ever used in your repo (incase you ever wanted to build an older version of your app), it is impossible to know if you are currently trying to build with dep@v1,v2, or v3 from just `go.sum`. `go.mod` is where you would get that information.